ui/src/components/build/create_build_config/BuildSection.js

```diff
@@ -0,0 +1,188 @@
+      <FormGroup>
+        <ControlLabel>Environment Variables</ControlLabel>
+        <KeyValueEditor onChange={rows => onEnvVarsChange(configState, rows)} />
+      </FormGroup>
+    </div>
+  );
+}
+
+function renderAndroidCredentials(component) {
+  const { androidCredentialsState } = component;
+  return (
+    <div>
+      <FormGroup validationState={androidCredentialsState.getValidationState(KEY_CR_BUILD_ANDROID_CREDENTIALS_NAME)}>
+        <ControlLabel className="required">Name</ControlLabel>
+        <FormControl
+          type="text"
+          onChange={e => androidCredentialsState.set(KEY_CR_BUILD_ANDROID_CREDENTIALS_NAME, e.target.value)}
+          value={androidCredentialsState.get(KEY_CR_BUILD_ANDROID_CREDENTIALS_NAME)}
+        />
+        <HelpBlock>A name for the credentials.</HelpBlock>
+      </FormGroup>
+      <FormGroup
+        validationState={androidCredentialsState.getValidationState(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE)}
+      >
+        <ControlLabel className="required">Android Keystore</ControlLabel>
+        <UploadControl
+          onTextLoaded={text =>
+            androidCredentialsState.set(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE, Buffer.from(text).toString('base64'))
+          }
+        />
+        <HelpBlock>Password protected PKCS12 file containing a key protected by the same password.</HelpBlock>
+      </FormGroup>
+      <FormGroup
+        validationState={androidCredentialsState.getValidationState(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_PASSWORD)}
+      >
+        <ControlLabel className="required">Android Keystore Password</ControlLabel>
+        <FormControl
+          type="password"
+          onChange={e =>
+            androidCredentialsState.set(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_PASSWORD, e.target.value)
+          }
+          value={androidCredentialsState.get(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_PASSWORD)}
+        />
+        <HelpBlock>Password for the PKCS12 archive and key.</HelpBlock>
+      </FormGroup>
+      <FormGroup
+        validationState={androidCredentialsState.getValidationState(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_ALIAS)}
+      >
+        <ControlLabel className="required">Keystore Alias</ControlLabel>
+        <FormControl
+          type="input"
+          onChange={e => androidCredentialsState.set(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_ALIAS, e.target.value)}
+          value={androidCredentialsState.get(KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_ALIAS)}
+        />
+        <HelpBlock>
+          The entry name of the private key/certificate chain you want to use to sign your APK(s). This entry must exist
+          in the key store uploaded. If your key store contains only one key entry, which is the most common case, you
+          can leave this field blank.
+        </HelpBlock>
+      </FormGroup>
+    </div>
+  );
+}
+
+function renderIOSCredentials(component) {
```

Comment:
the same with this one, it should be a component of its own.

Reply:
done